### DISCUSSION TITLE
Tracking when content was last changed

### DISCUSSION THREAD

**Discussion by cpinflux** [reclosedev/requests-cache#972] [General]
Hi, is it possible to know when the last time a frequently requested / cached response was last actually found to be different from the expired cache upon update? From my reading of the docs, the various useful fields added to the cache data don't actually define this? I can imagine it's something that's not worth the effort to deduce if it's not provided by the source's own modified headers etc.

**Comment by JWCook** [comment]
This info isn't currently available, but a `content_changed` response attribute could be added (https://github.com/requests-cache/requests-cache/issues/744). The easiest change to make would be setting this based on the results of a conditional request (200 vs 304).

For all other requests, we'd have to compare response contents after refreshing an expired response. My only concern is that in extreme cases (high request rate and large responses), that could result in a bit more CPU usage. In most cases, though, we'd be talking about a few nanoseconds. The worst case could be sped up a bit by saving a content hash in the cache, and checking a few other headers first like `Content-Length`.
